app/services/translation_provider.py

```
@@ -0,0 +1,184 @@
+            base_score = 0.9
+        
+        return base_score
+    
+    def optimize_prompt_for_provider(
+        self, text: str, source_lang: str, target_lang: str, context: Optional[str] = None
+    ) -> str:
+        """Optimize translation prompt for the specific provider."""
+        lang_names = {'en': 'English', 'ar': 'Arabic', 'bs': 'Bosnian'}
+        source_name = lang_names.get(source_lang, source_lang.upper())
+        target_name = lang_names.get(target_lang, target_lang.upper())
+        
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)